### PR TITLE
Adding Vue 3 Composition API

### PR DESCRIPTION
**Pull request by RiFrost**
### Description
To make the difference between Vue 2 and Vue 3 even clearer, perhaps using the Composition API would be a good idea. I stumbled across this because I only started learning Vue with version 3.x, so the Composition API is more familiar to me than the Vue 2 approach. I could imagine that people often feel this way and therefore maybe both approaches could be covered to save time in translation. 

Since this is my first real PR in an open source project, I'm not sure if this is a good idea and if this is the way to do it. 

I am open for criticism and feedback also regarding the PR.

### CODE REVIEW

File: demo/vue3js.html [hunk 1]
```diff
@@ -1,88 +1,97 @@
...
-            };
-            node.id = node.content = String(this.count++);
-            this.grid.addWidget(node);
-          },
-        },
-      }).mount("#app");
-    </script>
-  </body>
-</html>
+<!DOCTYPE html>
```
Comment (adumesny) [inline demo/vue3js.html:1]: can you make the indentation (2 spaces) and line ending the same so we can see what changed ? thanks

**Comment by RiFrost** [comment]
I dont think that this what i did will help you. I am sorry for that. But I think there is the Option to not show Whitespace changes when comparing files which should also resolve in seeing the actual changes better.

If this is not an Option i could make a new PR or otherwise you may help me sorting this out the right way. Sorry ^^

File: demo/vue3js.html [hunk 2]
```diff
@@ -1,88 +1,97 @@
...
+      <p>
+        If your app requires more complex render logic than the inline template
+        in `addWidget`, consider
+        <a
+          href="https://github.com/gridstack/gridstack.js/tree/develop/doc#makewidgetel"
+          >makeWidget</a
+        >
+        to let Vue deal with DOM rendering.
+      </p>
+      <button type="button" @click="addNewWidget()">Add Widget</button> {{ info
```
Comment (adumesny) [inline demo/vue3js.html:28]: not a useful change. please revert

File: demo/vue3js.html [hunk 3]
```diff
@@ -1,88 +1,97 @@
...
+    </main>
+    <script type="module">
+      import { createApp } from "https://cdn.jsdelivr.net/npm/vue@3.0.0/dist/vue.esm-browser.js";
+      import { ref, onMounted } from "vue";
+
+      createApp({
+        setup() {
+          let count = ref(0);
+          let info = ref("");
+          let grid = null;
```
Comment (adumesny) [inline demo/vue3js.html:40]: did you look at this and is it not an issue anymore ? great if this works as expected now.

// grid: undefined, // <-- don't do that. see https://github.com/gridstack/gridstack.js/issues/1635

File: demo/vue3js.html [hunk 4]
```diff
@@ -1,88 +1,97 @@
...
+          const items = [
+            { x: 2, y: 1, h: 2 },
+            { x: 2, y: 4, w: 3 },
+            { x: 4, y: 2 },
+            { x: 3, y: 1, h: 2 },
+            { x: 0, y: 6, w: 2, h: 2 },
+          ];
+
+          onMounted(() => {
+            grid = GridStack.init({
```
Comment (adumesny) [inline demo/vue3js.html:50]: no this.grid needed ? how is that scoped then ????
(don't want it global but this Vue component...)

I would insert the comment back
// Provides access to the GridStack instance across the Vue component.

**Review by adumesny — CHANGES_REQUESTED** [review]
yes I can merge and skip white spacing but still shows newline ending ?

Also best if you can merge your changes - git rebase -i HEAD~3  (I can squash on submit otherwise)

File: demo/vue3js.html (hunk 4 above)
Comment (RiFrost) [reply demo/vue3js.html:50]: From my understanding you got access to the grid within the setup function at all time. So it is only accessible for the Vue component itself. 
So you are able to define functions or vars etc. inside the setup function which are able to work with the grid.

File: demo/vue3js.html (hunk 3 above)
Comment (RiFrost) [reply demo/vue3js.html:40]: Since it is all inside the setup funtion i have not had any issues with that.

File: demo/vue3js.html (hunk 4 above)
Comment (adumesny) [reply demo/vue3js.html:50]: ok, did you want to make line ending change and merge into 1 review, then I can accept it. thank you!

**Pull request closed without merging**